src/Agent.js

```diff
@@ -146,6 +149,11 @@ export default class Agent {
 		if (context) {
 			this._applyContext(req, context);
 		}
+		if (Object.keys(headers).length) {
+			for (const key of Object.keys(headers)) {
+				req.set(key, headers[key]);
+			}
+		}
```

Comment:
i think you could just do:

```js
if (headers){
    req.set(headers);
}
```

([docs](https://visionmedia.github.io/superagent/#setting-header-fields))